.github/ISSUE_TEMPLATE/pip.md

```diff
@@ -0,0 +1,151 @@
+---
+name: PIP
+about: Submit a Pulsar Improvement Proposal (PIP)
+title: 'PIP-XYZ: '
+labels: PIP
+---
+
+<!--
+RULES
+* Never place a link to an external site like Google Doc. The proposal should be in this issue entirely.
+* Use a spelling and grammar checker tools if available for you (there are plenty of free ones)
+
+PROPOSAL HEALTH CHECK
+I can read the design document and understand the problem statement and what you plan to change *without* resorting to a couple of hours of code reading just to start having a high level understanding of the change.
+-->
+
+# Background knowledge
+
+<!--
+Describes all the knowledge you need to know in order to understand all the other sections in this PIP
+
+* Give a high level explanation on all concepts you will be using throughout this document. For example, if you want to talk about Persistent Subscriptions, explain briefly (1 paragraph) what this is. If you're going to talk about Transaction Buffer, explain briefly what this is. 
+  If you're going to change something specific, then go into more detail about it and how it works. 
+* Provide links where possible if a person wants to dig deeper into the background information. 
+
+DON'T
+* Do not include links *instead* explanation. Do provide links for further explanation.
+
+EXAMPLES
+* See PIP-248, Background section to get an understanding on how you add the background knowledge needed.
+  (They also included the motivation there, but ignore it as we place that in Motivation section explicitly)
+-->
+
+# Motivation
+
+<!--
+Describe the problem this proposal is trying to solve.
+
+* Explain what is the problem you're trying to solve - current situation.
+* This section is the "Why" of your proposal.
+-->
+
+# Goals
+
+## In Scope
```

Comment:
Maybe use "Goals" and "Non-Goals" to reduce hierarchy?

Reply:
Goals is what you want to achieve.
Scope - what you can actually do in the PIP
Out of scope - you don't have time , so it is not included in the pip at all.

Non-goals: Objectives you will achieve in this pip (in scope) but are not your primary goals.

I think it has different semantics.